include/pure_pursuit.h

```diff
@@ -57,6 +58,218 @@ class PurePursuit{
+					currentLoc.y = (currentLoc.y + slope);
+
+					//if currentLocation goes beyond next point, it is set to the next point
+					if (currentLoc.x >= m_robot_path[i + 1].x) {
+						currentLoc.x = m_robot_path[i + 1].x;
+						currentLoc.y = m_robot_path[i + 1].y;
+					}
+					if (currentLoc.y >= m_robot_path[i + 1].y) {
+						currentLoc.x = m_robot_path[i + 1].x;
+						currentLoc.y = m_robot_path[i + 1].y;
+					}
+
+					//calculates distance to robot
+					distanceToRobot = sqrt(pow(abs(-currentLoc.x + state.x), 2) + pow(abs(-currentLoc.y + state.y), 2));
+
+					//checks to see if currentLocation is closer to the current position
+					if (distanceToRobot < shortestDistance) {
+						shortestDistance = distanceToRobot;
+						closestLocation = currentLoc;
+					}
+				}
+				//segment is going up and to the left
+				else if ((m_robot_path[i].y < m_robot_path[i + 1].y) && (m_robot_path[i].x > m_robot_path[i + 1].x)) {
+					counter--;
+
+					currentLoc.x = (m_robot_path[i].x + counter);
+					currentLoc.y = (currentLoc.y + -slope);
+
+					//if currentLocation goes beyond next point, it is set to the next point
+					if (currentLoc.x <= m_robot_path[i + 1].x) {
+						currentLoc.x = m_robot_path[i + 1].x;
+						currentLoc.y = m_robot_path[i + 1].y;
+					}
+					if (currentLoc.y >= m_robot_path[i + 1].y) {
+						currentLoc.x = m_robot_path[i + 1].x;
+						currentLoc.y = m_robot_path[i + 1].y;
+					}
+
+					//calculates distance to robot
+					distanceToRobot = sqrt(pow(abs(-currentLoc.x + state.x), 2) + pow(abs(-currentLoc.y + state.y), 2));
+
+					//checks to see if currentLocation is closer to the current position
+					if (distanceToRobot < shortestDistance) {
+						shortestDistance = distanceToRobot;
+						closestLocation = currentLoc;
+					}
+				}
+				//segment is going down and to the left
+				else if ((m_robot_path[i].y > m_robot_path[i + 1].y) && (m_robot_path[i].x > m_robot_path[i + 1].x)) {
+					counter--;
+
+					currentLoc.x = (m_robot_path[i].x + counter);
+					currentLoc.y = (currentLoc.y + -slope);
+
+					//if currentLocation goes beyond next point, it is set to the next point
+					if (currentLoc.x <= m_robot_path[i + 1].x) {
+						currentLoc.x = m_robot_path[i + 1].x;
+						currentLoc.y = m_robot_path[i + 1].y;
+					}
+					if (currentLoc.y <= m_robot_path[i + 1].y) {
+						currentLoc.x = m_robot_path[i + 1].x;
+						currentLoc.y = m_robot_path[i + 1].y;
+					}
+
+					//calculates distance to robot
+					distanceToRobot = sqrt(pow(abs(-currentLoc.x + state.x), 2) + pow(abs(-currentLoc.y + state.y), 2));
+
+					//checks to see if currentLocation is closer to the current position
+					if (distanceToRobot < shortestDistance) {
+						shortestDistance = distanceToRobot;
+						closestLocation = currentLoc;
+					}
+				}
+				//segment is going down and to the right
+				else if ((m_robot_path[i].y > m_robot_path[i + 1].y) && (m_robot_path[i].x < m_robot_path[i + 1].x)) {
```

Comment:
when using vectors dont use [i] to access elements, use std::vector::at. Ex: m_robot_path.at(i) instead of m_robot_path[i]

Reply:
Do this with every occurrence of [] brackets